plugins/registry.js

```diff
@@ -461,4 +461,23 @@ export default class PluginRegistry {
 
         return {id, showRHSPlugin: showRHSPlugin(id)};
     }
+
+    // Register a component to replace code blocks with components provided by plugins.
+    // Accepts a React component. Returns a unique identifier.
+    registerCodeBlockComponent(component, languages) {
```

Comment:
I'm wondering if the parameters should be reversed here, e.g. like `registerFilePreviewComponent`. If `languages` were a property of the component, I'd say this is fine, but I think we're overriding languages /with/ this component.

Reply:
I'm sorry, but it's not clear for me. How we can override languages here?

Reply:
Apologies, I'm wondering if we should just declare the signature of this API as:
```suggestion
    registerCodeBlockComponent(languages, component) {
```

to match how other functions arrange their parameters (e.g. `registerFilePreviewComponent`).

Reply:
Oh, I see now. Indeed, this is a good suggestion! Done :+1:.